pkg/controller/acm/controller.go

```diff
@@ -0,0 +1,266 @@
+
+	cr.Status.AtProvider = acm.GenerateCertificateStatus(certificate)
+
+	tags, err := e.client.ListTagsForCertificateRequest(&awsacm.ListTagsForCertificateInput{
+		CertificateArn: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(acm.IsErrorNotFound, err), errListTagsFailed)
+	}
+
+	return managed.ExternalObservation{
+		ResourceUpToDate: acm.IsCertificateUpToDate(cr.Spec.ForProvider, certificate, tags.Tags),
+		ResourceExists:   true,
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+
+	cr, ok := mgd.(*v1alpha1.Certificate)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	response, err := e.client.RequestCertificateRequest(acm.GenerateCreateCertificateInput(meta.GetExternalName(cr), &cr.Spec.ForProvider)).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+	}
+
+	meta.SetExternalName(cr, aws.StringValue(response.RequestCertificateOutput.CertificateArn))
+	return managed.ExternalCreation{}, errors.Wrap(e.kube.Update(ctx, cr), errPersistExternalName)
+
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+
+	cr, ok := mgd.(*v1alpha1.Certificate)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	// Update Certificate tags
+	if len(cr.Spec.ForProvider.Tags) > 0 {
+
+		desiredTags := make([]awsacm.Tag, len(cr.Spec.ForProvider.Tags))
+		for i, t := range cr.Spec.ForProvider.Tags {
+			desiredTags[i] = awsacm.Tag{Key: aws.String(t.Key), Value: aws.String(t.Value)}
+		}
+
+		currentTags, err := e.client.ListTagsForCertificateRequest(&awsacm.ListTagsForCertificateInput{
+			CertificateArn: aws.String(meta.GetExternalName(cr)),
+		}).Send(ctx)
+
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(acm.IsErrorNotFound, err), errListTagsFailed)
+		}
+
+		if len(desiredTags) != len(currentTags.Tags) {
+			_, err := e.client.RemoveTagsFromCertificateRequest(&awsacm.RemoveTagsFromCertificateInput{
+				CertificateArn: aws.String(meta.GetExternalName(cr)),
+				Tags:           currentTags.Tags,
+			}).Send(ctx)
+			if err != nil {
+				return managed.ExternalUpdate{}, errors.Wrap(err, errRemoveTagsFailed)
+			}
+		}
+		_, err = e.client.AddTagsToCertificateRequest(&awsacm.AddTagsToCertificateInput{
+			CertificateArn: aws.String(meta.GetExternalName(cr)),
+			Tags:           desiredTags,
+		}).Send(ctx)
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errAddTagsFailed)
+		}
+	}
+
+	// Update the Certificate Option
+	if aws.StringValue(cr.Spec.ForProvider.CertificateAuthorityARN) == "" && cr.Spec.ForProvider.CertificateTransparencyLoggingPreference != nil {
```

Comment:
Can you only update public certificates?

Reply:
I missed removing the check for CertificateAuthorityARN. We can update both the Private and Public certificate. 
Updated the check now and can update both certificate Options.